com.unity.multiplayer.mlapi/Runtime/Core/NetworkManager.cs

```diff
@@ -249,39 +247,58 @@ private void OnValidate()
                 };
             }
 
+            // During OnValidate we will always clear out NetworkPrefabOverrideLinks and rebuild it
+            NetworkConfig.NetworkPrefabOverrideLinks.Clear();
+
+            // Check network prefabs and assign to dictionary for quick look up
             for (int i = 0; i < NetworkConfig.NetworkPrefabs.Count; i++)
             {
                 if (NetworkConfig.NetworkPrefabs[i] != null && NetworkConfig.NetworkPrefabs[i].Prefab != null)
                 {
-                    if (NetworkConfig.NetworkPrefabs[i].Prefab.GetComponent<NetworkObject>() == null)
+                    var networkObject = NetworkConfig.NetworkPrefabs[i].Prefab.GetComponent<NetworkObject>();
+                    if (networkObject == null)
                     {
                         if (NetworkLog.CurrentLogLevel <= LogLevel.Normal)
                         {
                             NetworkLog.LogWarning($"{nameof(NetworkPrefab)} [{i}] does not have a {nameof(NetworkObject)} component");
                         }
                     }
-                }
-            }
+                    else
+                    {
+                        // Defautlt to the standard NetworkPrefab.Prefab's NetworkObject first
+                        var globalObjectIdHash = networkObject.GlobalObjectIdHash;
 
-            int playerPrefabCount = NetworkConfig.NetworkPrefabs.Count(x => x.IsPlayer);
+                        // Now check to see if it has an override
```

Comment:
```suggestion
```

over-commenting, isn't it obvious?

Reply:
Actually, comments help guide people as to what is happening.  
While it is obvious to us, this level of commenting to someone who is completely new to the SDK is actually useful.
I would rather leave comments in than remove them.

Reply:
I disagree :P Let's discuss internally :)